pkg/helper/secretutils.go

```diff
@@ -74,3 +77,89 @@ func MergeSecretData(from, to map[string][]byte) map[string][]byte {
 
 	return result
 }
+
+type SecretCacheElement struct {
+	Secret *v1.Secret
+	Err    error
+}
+
+type SecretSource struct {
+	client      client.Client
+	namespace   string
+	secretCache *MemoryCache
+}
+
+func NewSecretSource(client client.Client, namespace string) *SecretSource {
+	return &SecretSource{
+		client:      client,
+		namespace:   namespace,
+		secretCache: NewMemoryCache(),
+	}
+}
+
+func (s *SecretSource) FieldValue(secretName, fieldName string, def string) (string, error) {
+	return s.fieldReader(secretName, fieldName, false, false, def)
+}
+
+func (s *SecretSource) FieldValueFromRequiredSecret(secretName, fieldName string, def string) (string, error) {
+	return s.fieldReader(secretName, fieldName, true, false, def)
+}
+
+func (s *SecretSource) RequiredFieldValue(secretName, fieldName string) (string, error) {
```

Comment:
Does this method has any use case?
In case a field of a secret is required, shouldn't the secret be required too? If a field is required (there's no default value) and the secret does not exist wouldn't we always get an error?

Reply:
Yes, 

The field requirement implies secret existence.

I will remove the method.